### ISSUE TITLE
[FE] Refactor code to have the List and Questionnaire be in same front end app. 

### ISSUE THREAD

**Issue by mdewey** [department-of-veterans-affairs/va.gov-team#18331]
## User Story

For a veteran, 
The questionnaire app should be one front app application 
so that the veteran has an optimized experience. 

## Related Documentation
 - [MVP: Visit Intro Initiative (Proof of Concept)](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/health-care/questionnaire/product/initiatives/visit-intro.md)

## Description
Currently, the List and the form are two different front end applications. The FE teams need to combine the two apps into one. This is so the List and the form can share data and reduce calls to the API. Not only does this follow best practices for our current tech stack, but this will also gain us better performance and reduce the work the back end teams needs to do. 

## Tasks
> this is a list of technical tasks that are needed for this ticket 
- [ ] Update the apps to be able to share data in Redux
- [ ] Update user tests
- [ ] Update e2e tests 
- [ ] Update relevant documentations

## Acceptance Criteria
- [ ] Tests should all still pass
- [ ] Passes Accessability checks using axe
- [ ] PR has been merged into master

**Comment by mdewey** [comment]
did: Have prototype created with dummy form. https://github.com/department-of-veterans-affairs/vets-website/pull/16075

TODO: 
- [ ] move over real form
- [ ] refactor shared codepull in selected appointment from redux
- [ ] check direct load. 
- [ ] Update tests
- [ ] submit PR

**Comment by mdewey** [comment]
Current status. 

After running into several issues, I reached out to platform for guidance. 

[Here is the slack thread.](https://dsva.slack.com/archives/CBU0KDSB1/p1613588539283500) 

**TL;DR;** The first recommended way of sharing data between a non-form app and a form app is to keep the apps separate and use session storage. 

This is an okay solution for now. The platform team is open to a better solution in the long term but lacks the bandwidth and knowledge to make better recommendations. I will look into possible solutions. 

**longer answer**

The core issue is that the forms library is not flexible to pass data to a form from another app and is hard at best to combine with another application. Other teams have hacked around this or used methods that do not work for us, such as API calls or the `additioanlPages` option on the `formConfig`. Just to note, VOAS got around this by not using the forms system library, but duplicating code and using their own implementations. Post MVP, depending on requirements we might need to go that route. 

The plan: 
When a user clicks an appointment to continue, the list will store that appointment in session storage.

The form, instead of going to the API, will look in the sessions storage for appointment details 

If no appointment details are found, then redirect back to the list page. 

We will store appointment details client-side for 24 hrs?
